### ISSUE TITLE
Compilation fails on Windows OSs

### ISSUE THREAD

**Issue by dschlaep** [Burke-Lauenroth-Lab/rSOILWAT2#34]
From [r-hub builder](https://builder.r-hub.io):

```
 507#> Warning: running command '"C:/PROGRA~1/R/R-34~1.1/bin/x64/Rcmd.exe" INSTALL -l "C:/Users/USERNOhuaCzcbL/R" --no-html "C:\Users\USERNO~1\RSOILW~1.RCH\00_PKG~1\RSOILW~1"' had status 1
 508#> * checking whether package 'rSOILWAT2' can be installed ... ERROR
 509#> Installation failed.
 510#> See 'C:/Users/USERNOhuaCzcbL/rSOILWAT2.Rcheck/00install.out' for details.
 511#> * DONE
 512#> Status: 1 ERROR, 1 WARNING
```

**Comment by dschlaep** [comment]
From Rachel:

```
tools::Rcmd(args = paste("INSTALL rSOILWAT2"))
* installing to library 'C:/Users/XXXXXXX/Documents/R/win-library/3.4'
* installing *source* package 'rSOILWAT2' ...
** libs

*** arch - i386
Warning: running command 'make -f "Makevars" -f "C:/PROGRA~1/R/R-34~1.1/etc/i386/Makeconf" -f "C:/PROGRA~1/R/R-34~1.1/share/make/winshlib.mk" SHLIB="rSOILWAT2.dll" OBJECTS="SW_Control.o SW_Files.o SW_Flow.o SW_Flow_lib.o SW_Main.o SW_Main_Function.o SW_Markov.o SW_Model.o SW_Output.o SW_R_lib.o SW_Site.o SW_Sky.o SW_SoilWater.o SW_VegEstab.o SW_VegProd.o SW_Weather.o Times.o filefuncs.o generic.o mymemory.o rands.o"' had status 127
ERROR: compilation failed for package 'rSOILWAT2'
* removing 'C:/Users/Rachel Renne/Documents/R/win-library/3.4/rSOILWAT2'
Warning message:
running command '"C:/PROGRA~1/R/R-34~1.1/bin/x64/Rcmd.exe" INSTALL rSOILWAT2' had status 1
```

**Comment by dschlaep** [comment]
From r-hub builder rSOILWAT2.Rcheck/00install.out:

```
C:\Rtools34\mingw_32\bin\nm.exe: SW_R_lib.o: File format not recognized
C:/Rtools34/mingw_32/bin/gcc -shared -s -static-libgcc -o rSOILWAT2.dll tmp.def SW_Control.o SW_Files.o SW_Flow.o SW_Flow_lib.o SW_Main.o SW_Main_Function.o SW_Markov.o SW_Model.o SW_Output.o SW_R_lib.o SW_Site.o SW_Sky.o SW_SoilWater.o SW_VegEstab.o SW_VegProd.o SW_Weather.o Times.o filefuncs.o generic.o mymemory.o rands.o -Ld:/Compiler/gcc-4.9.3/local330/lib/i386 -Ld:/Compiler/gcc-4.9.3/local330/lib -LC:/PROGRA~1/R/R-34~1.1/bin/i386 -lR
SW_R_lib.o: file not recognized: File format not recognized
collect2.exe: error: ld returned 1 exit status
C:\Rtools33\GCC-46~1.3\bin\nm.exe: C:/Users/USERNOhuaCzcbL/rSOILWAT2.Rcheck/00_pkg_src/rSOILWAT2/src-i386/SW_R_lib.o: File format not recognized
Warning message:
running command '"C:\Rtools33\GCC-46~1.3\bin\nm.exe" -Pg "C:/Users/USERNOhuaCzcbL/rSOILWAT2.Rcheck/00_pkg_src/rSOILWAT2/src-i386/SW_R_lib.o"' had status 1 
no DLL was created
ERROR: compilation failed for package 'rSOILWAT2'
* removing 'C:/Users/USERNOhuaCzcbL/R/rSOILWAT2'
```

**Comment by dschlaep** [comment]
> Under Windows the same commands work, but Makevars.win will be used in preference to Makevars, and only src/Makefile.win will be used by R CMD INSTALL with src/Makefile being ignored. 

https://cran.r-project.org/doc/manuals/R-exts.html#Creating-shared-objects

**Comment by Zachary-Kramer** [comment]
I use Windows 10 64-bit and compilation succeeds.

![image](https://user-images.githubusercontent.com/16931318/28218392-4171e454-686d-11e7-811f-b48a72fc4ea4.png)

However, I enabled 32-bit compilation to see if that would cause this problem, and although the compilation passed, the loading failed:
![image](https://user-images.githubusercontent.com/16931318/28218499-98b5c7da-686d-11e7-82b7-3bbbd560bedd.png)

Is that a separate issue?

**Comment by dschlaep** [comment]
@Zachary-Kramer Many thanks for testing this on your Windows 10!

It seems that the make call of the `R CMD INSTALL` sees the 'rSOILWAT2.dll' and doesn't re-compile it. I guess that the 32-bit compilation fails because the version of 'rSOILWAT2.dll' that is already there is 64-bit? So I am not sure that this is a separate issue.

I would be interested to know how you succeeded in compiling 'rSOILWAT2.dll'.

**Comment by dschlaep** [comment]
Updating SOILWAT2's makefile and making sure that no lingering .so files hang around (commit https://github.com/Burke-Lauenroth-Lab/SOILWAT2/commit/97a7dc167492de2e1bdf53b2e22ffb3ce3fbddf1), allows to compile on r-hub for 'Windows Server 2008 R2 SP1, R-release, 32/64 bit':

```
 953#> installing to C:/Users/USERCBcHSEQeaJ/R/rSOILWAT2/libs/x64
 954#> ** R
 955#> ** data
 956#> *** moving datasets to lazyload DB
 957#> ** inst
 958#> ** byte-compile and prepare package for lazy loading
 959#> ** help
 960#> *** installing help indices
 961#> ** building package indices
 962#> ** testing if installed package can be loaded
 963#> *** arch - i386
 964#> *** arch - x64
 965#> * MD5 sums
 966#> packaged installation of 'rSOILWAT2' as rSOILWAT2_1.2.13.zip
 967#> * DONE (rSOILWAT2)
```

**Comment by dschlaep** [comment]
Close #34
This issue was fixed before or with commit c648a109b944b4a20ab26cf3498ce5cae1871851.

However, appveyor still reports failure because unit tests produce errors whereas the compilation itself is successful.
